CHANGELOG.md

```diff
@@ -1,5 +1,63 @@
 # go-ipfs changelog
 
+### 0.4.2 - 2016-05-17
+
+This is a patch release which fixes perfomance and networking bugs in go-libp2p,
+You should see improvements in CPU and RAM usage, as well as speed of object lookups.
+There are also a few other nice improvements.
+
+* Notable Fixes
+  * Set a deadline for dialing attempts. This prevents a node from accumulating
+    failed connections. (@whyrusleeping)
+  * Avoid unneccessary string/byte conversions in go-multihash. (@whyrusleeping)
+  * Fix a deadlock around the yamux stream muxer. (@whyrusleeping)
+  * Fix a bug that left channels open, causing hangs. (@whyrusleeping)
+  * Fix a bug around yamux which caused connection hangs. (@whyrusleeping)
+
+* Enhancements
+  * Add NetBSD support. (@erde74)
+  * Set Cache-Control: immutable on /ipfs responses. (@kpcyrd)
+  * Have `ipfs init` optionally accept a default configuration from stdin. (@sivachandran)
+  * Add `ipfs log ls` command for listing logging subsystems. (@hsanjuan)
+  * Allow bitswap to read multiple messages per stream. (@whyrusleeping)
+  * Remove `make toolkit_upgrade` step. (@chriscool)
+
+* Documentation
+  * Add a debug-guidelines document. (@richardlitt)
+  * Update the contribute document. (@richardlitt)
+  * Fix documentation of all `ipfs` commands. (@richardlitt)
```

Comment:
Maybe it should be split into "Fix default values of many `ipfs` commands" and "Improve commands documentation"?